### PR TITLE
feat: Allow for dynamic and static height Modals

### PR DESCRIPTION
**Pull request by bdow**
Also adding type of `IconKey`

### CODE REVIEW

File: src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -35,6 +35,7 @@ export interface ModalProps {
  */
 export function Modal(props: ModalProps) {
   const { title, size = "md", content, forceScrolling } = props;
+  const isFixedHeight = typeof size !== "string";
```
Comment (stephenh) [inline src/components/Modal/Modal.tsx:38]: Won't the `size = md` right before this throw off the "default is dynamic"?

Comment (stephenh) [reply src/components/Modal/Modal.tsx:38]: Oh, this is "the default _width_" is will `md`, but height will default..

Comment (bdow) [reply src/components/Modal/Modal.tsx:38]: I don't think so? Maybe I'm missing something?
The height will be dynamic if the `size` attribute is defined as a `ModalSize`. If no size is passed in, then it defaults to a ModalSize and that allow for dynamic by default.
If `size` is not a `string`, meaning it has been defined as an `object`, with `width` and `height`, that is when we treat it as `isFixedHeight`.

Comment (bdow) [reply src/components/Modal/Modal.tsx:38]: Right right right. Okay. Will add in the comment.

Comment (stephenh) [reply src/components/Modal/Modal.tsx:38]: Yep, I get it now

**Review by stephenh — APPROVED** [review]

**Pull request merged**